src/plugins/fs/mount.c

```diff
@@ -343,6 +343,9 @@ static gboolean get_mount_error_new (struct libmnt_context *cxt, int rc, MountAr
         if (permission)
             g_set_error (error, BD_FS_ERROR, BD_FS_ERROR_AUTH,
                          "Operation not permitted.");
+        else if (syscall_errno == ENODEV)
+            g_set_error (error, BD_FS_ERROR, BD_FS_ERROR_NOFS,
```

Comment:
I was not sure whether to introduce a new error code, kinda liked the existing `BD_FS_ERROR_NOFS`.

Reply:
A new error code sounds better. Maybe you could add the new error code only on master and use `BD_FS_ERROR_NOFS` on 2.x?

Reply:
Not an option, I might need it for a stable udisks branch. Things will need to be consistent.

Reply:
I meant using `BD_FS_ERROR_NOFS` with libblockdev 2.x and using the new error code with 3.0. To avoid ABI changes and new error code on "stable" libblockdev. But if you think this is better, I'm ok with it.

Reply:
When backporting to 2.x branch, the new error code would need to be specified as `BD_FS_ERROR_UNKNOWN_FS = 12` to maintain code numbers.

Reply:
> When backporting to 2.x branch, the new error code would need to be specified as `BD_FS_ERROR_UNKNOWN_FS = 12` to maintain code numbers.

Please prepare the 2.x backport PR too, thanks. We already have quite a few bugfixes there so I plan to do a new release soon.